File: mobile/lib/services/payment_session_context.dart
import '../utils/id.dart';

/// سياق جلسة المستخدم التي تُسند إليها المدفوعات الجديدة.
///
/// لا يمثل هذا جدول نوبات مستقلاً؛ بل يمثل جلسة تسجيل الدخول الحالية
/// كما اتُّفق عليه في الخيار A. يُعاد توليد sessionUuid عند كل login/restore.
class PaymentSessionContext {
  PaymentSessionContext._();

  static int? userId;
  static String? userName;
  static String? sessionUuid;
  static String? cloudUserId;
  static DateTime? startedAt;

  static bool get isActive => userId != null && sessionUuid != null;

  static void start({
    required int userId,
    required String userName,
    String? sessionUuid,
    String? cloudUserId,
    DateTime? startedAt,
  }) {
    PaymentSessionContext.userId = userId;
    PaymentSessionContext.userName = userName;
    PaymentSessionContext.sessionUuid = sessionUuid ?? IdGen.uuid();
    PaymentSessionContext.cloudUserId = cloudUserId;
    PaymentSessionContext.startedAt = startedAt ?? DateTime.now();
  }

  static void clear() {
    userId = null;
    userName = null;
    sessionUuid = null;
    cloudUserId = null;
    startedAt = null;
  }
}

File: mobile/android/app/src/main/kotlin/com/marina/marina/domain/session/PaymentSessionContext.kt
package com.marina.marina.domain.session

import java.util.UUID

/**
 * Payment-session context that new payments are attributed to.
 *
 * Ported 1:1 from the Flutter app's `lib/services/payment_session_context.dart`.
 *
 * This is NOT a standalone shifts table; it represents the current login
 * session (Option A as agreed in the Flutter app). The [sessionUuid] is
 * regenerated on every login/restore.
 */
object PaymentSessionContext {

    var userId: Long? = null
        private set

    var userName: String? = null
        private set

    var sessionUuid: String? = null
        private set

    var cloudUserId: String? = null
        private set

    var startedAt: Long? = null
        private set

    val isActive: Boolean
        get() = userId != null && sessionUuid != null

    fun start(
        userId: Long,
        userName: String,
        sessionUuid: String? = null,
        cloudUserId: String? = null,
        startedAt: Long? = null
    ) {
        this.userId = userId
        this.userName = userName
        this.sessionUuid = sessionUuid ?: UUID.randomUUID().toString()
        this.cloudUserId = cloudUserId
        this.startedAt = startedAt ?: System.currentTimeMillis()
    }

    fun clear() {
        userId = null
        userName = null
        sessionUuid = null
        cloudUserId = null
        startedAt = null
    }
}
